chunk/chunk.go

```diff
@@ -341,12 +348,11 @@ func (s *ValidatorStore) Put(ctx context.Context, mode ModePut, chs ...Chunk) (e
 // validate returns true if one of the validators
 // return true. If all validators return false,
 // the chunk is considered invalid.
-func (s *ValidatorStore) validate(ch Chunk) (bool, Type) {
+func (s *ValidatorStore) validate(ch Chunk) bool {
 	for _, v := range s.validators {
-		valid, chunkType := v.Validate(ch)
-		if valid {
-			return true, chunkType
+		if v.Validate(ch) {
```

Comment:
I would prefer that the chunk type is set here. currently it is in the respective validator package but the type is here anyway, and the type setting code is unnecassarily duplicated. 
there are various options here. I prefer one where the validators are registered with a struct that has type as a field and the  validate function the other field.

Reply:
completely agree on this one, so thanks for the suggestion.

i've added the `Type` func to the `Validator` interface for this purpose.